client/my-sites/pages/page-list.jsx

```diff
@@ -154,6 +168,35 @@ class Pages extends Component {
 		return markedPages;
 	}
 
+	updateShadowStatus = ( globalID, shadowStatus ) =>
+		new Promise( resolve =>
+			this.setState( state => {
+				if ( shadowStatus ) {
+					// add or update the `globalID` key in the `shadowItems` map
+					return {
+						shadowItems: {
+							...state.shadowItems,
+							[ globalID ]: shadowStatus,
+						},
+					};
+				}
+
+				// remove `globalID` from the `shadowItems` map
+				const newShadowItems = without( state.shadowItems, globalID );
+				const newState = {
+					shadowItems: newShadowItems,
+				};
+
+				// if the last shadow item just got removed, start showing the up-to-date post
+				// list as specified by props.
+				if ( size( newShadowItems ) === 0 ) {
+					newState.pages = this.props.pages;
```

Comment:
Are you sure it is okay to access `this.props` here? I always figured you were supposed to use the second argument to the setState updater fn.

Reply:
Good point, I wasn't aware that the `setState` updater function has a second `props` param. Pushed a fix.